--- a/src/corelib/global/qlibraryinfo.cpp
+++ b/src/corelib/global/qlibraryinfo.cpp
@@ -372,6 +372,8 @@ static QString getRelocatablePrefix(QLibraryInfoPrivate::UsageMode usageMode)
     const QString prefixDir = QString(libDirCFString) + "/" QT_CONFIGURE_LIBLOCATION_TO_PREFIX_PATH;

     prefixPath = QDir::cleanPath(prefixDir);
+    if (prefixPath.startsWith("/usr/local/Cellar/qt-base/"))
+      prefixPath = "/usr/local";
 #elif defined(Q_OS_WASM)
     // Emscripten expects to find shared libraries at the root of the in-memory
     // file system when resolving dependencies for for dlopen() calls. So that's
