--- a/src/corelib/global/qlibraryinfo.cpp
+++ b/src/corelib/global/qlibraryinfo.cpp
@@ -365,6 +365,9 @@ static QString getRelocatablePrefix(QLibraryInfoPrivate::UsageMode usageMode)
     const QString prefixDir = QString(libDirCFString) + "/" QT_CONFIGURE_LIBLOCATION_TO_PREFIX_PATH;
 
     prefixPath = QDir::cleanPath(prefixDir);
+    if (prefixPath.startsWith("/usr/local/Cellar/qt-base/")) {
+        prefixPath = "/usr/local";
+    }
 #elif QT_CONFIG(dlopen)
     Q_UNUSED(usageMode);
     Dl_info info;
